src/ImageService.Broker/Consumers/CreateImagesConsumer.cs

```diff
@@ -8,238 +11,190 @@
-                dbImageProject = _dbImageProjectMapper.Map(createImage);
-                resizedContent = _resizeHelper.Resize(createImage.Content, createImage.Extension);
-
-                if (string.IsNullOrEmpty(resizedContent))
-                {
-                    dbImageProject.ParentId = dbImageProject.Id;
-                    previewIds.Add(dbImageProject.Id);
-                }
-                else
-                {
-                    dbPrewiewImageProject = _dbImageProjectMapper.Map(createImage, dbImageProject.Id, resizedContent);
-                    dbImages.Add(dbPrewiewImageProject);
-                    previewIds.Add(dbPrewiewImageProject.Id);
-                }
-
-                dbImages.Add(dbImageProject);
-            }
-
-            if (_imageProjectRepository.Create(dbImages) == null)
-            {
-                return ICreateImagesResponse.CreateObj(null);
-            }
-
-            return ICreateImagesResponse.CreateObj(previewIds);
+          dbImageProject.ParentId = dbImageProject.Id;
+          previewIds.Add(dbImageProject.Id);
         }
-
-        private object CreateNewsImages(ICreateImageRequest request)
+        else
         {
-            if (request.Images == null)
-            {
-                return ICreateImagesResponse.CreateObj(null);
-            }
-
-            List<DbImageNews> dbImages = new();
-            List<Guid> previewIds = new();
-            DbImageNews dbImageNews;
-            DbImageNews dbPrewiewImageNews;
-            string resizedContent;
-
-            foreach (CreateImageData createImage in request.Images)
-            {
-                dbImageNews = _dbImageNewsMapper.Map(createImage);
-                resizedContent = _resizeHelper.Resize(createImage.Content, createImage.Extension);
-
-                if (string.IsNullOrEmpty(resizedContent))
-                {
-                    dbImageNews.ParentId = dbImageNews.Id;
-                    previewIds.Add(dbImageNews.Id);
-                }
-                else
-                {
-                    dbPrewiewImageNews = _dbImageNewsMapper.Map(createImage, dbImageNews.Id, resizedContent);
-                    dbImages.Add(dbPrewiewImageNews);
-                    previewIds.Add(dbPrewiewImageNews.Id);
-                }
-
-                dbImages.Add(dbImageNews);
-            }
-
-            if (_imageNewsRepository.Create(dbImages) == null)
-            {
-                return ICreateImagesResponse.CreateObj(null);
-            }
-
-            return ICreateImagesResponse.CreateObj(previewIds);
+          dbPrewiewImageProject = _dbImageProjectMapper.Map(createImage, dbImageProject.Id, resizedContent);
+          dbImages.Add(dbPrewiewImageProject);
+          previewIds.Add(dbPrewiewImageProject.Id);
         }
 
-        private object CreateMessageImages(ICreateImageRequest request)
+        dbImages.Add(dbImageProject);
+      }
+
+      if (_imageProjectRepository.Create(dbImages) == null)
```

Comment:
useless check. If repo return null you will create response with null, why don't you want to use this null like argument

Reply:
you have it in every method

Reply:
repo returns all ids - for preview and for full images, but we send back only preview ids
so, we cant just do "return repo.Create()"